l1-contracts/src/core/Rollup.sol

```diff
@@ -124,6 +118,38 @@ contract Rollup is IRollup {
       revert Errors.Rollup__InvalidSequencer(msg.sender);
     }
 
+    archive = _archive;
+    lastBlockTs = block.timestamp;
+
+    bytes32 inHash = INBOX.consume();
+    if (header.contentCommitment.inHash != inHash) {
+      revert Errors.Rollup__InvalidInHash(inHash, header.contentCommitment.inHash);
+    }
+
+    // TODO(#7218): Revert to fixed height tree for outbox, currently just providing min as interim
+    // Min size = smallest path of the rollup tree + 1
+    (uint256 min,) = MerkleLib.computeMinMaxPathLength(header.contentCommitment.numTxs);
+    uint256 l2ToL1TreeMinHeight = min + 1;
+    OUTBOX.insert(
+      header.globalVariables.blockNumber, header.contentCommitment.outHash, l2ToL1TreeMinHeight
+    );
+
+    // pay the coinbase 1 gas token if it is not empty and header.totalFees is not zero
+    if (header.globalVariables.coinbase != address(0) && header.totalFees > 0) {
+      GAS_TOKEN.transfer(address(header.globalVariables.coinbase), header.totalFees);
+    }
```

Comment:
Hmm, shouldn't this go out only after the block is proven (to avoid the block being reverted due to missing proof)? If that's the case, I'd add a note/todo about it

Reply:
It's one of the items I got on the design doc, yep. Fee payment will have to change quite a lot.